File: runtime/java/units/wasi/clock_time_get.java
// requires: memory/i64_store
int wasi_clock_time_get(int id, long precision, int outPtr) {
    switch (id) {
        case 0: // realtime
        case 1: // monotonic
        case 2: // process cputime
        case 3: // thread cputime
            memory.i64_store(
                Integer.toUnsignedLong(outPtr), System.currentTimeMillis() * 1_000_000L);
            return WASI_OK;
        default:
            return WASI_INVAL;
    }
}

File: runtime/go/units/wasi/clock_time_get.go
// requires: memory/i64_store
func (w *WASI) wasi_clock_time_get(id uint32, precision uint64, outPtr uint32) uint32 {
    switch id {
    case 0, 1, 2, 3: // realtime / monotonic / process / thread cputime
        w.memory.i64_store(uint64(outPtr), uint64(time.Now().UnixNano()))
    default:
        return wasiInval
    }
    return wasiOk
}
